pkg/image/registry/image/strategy.go

```diff
@@ -44,6 +45,25 @@ func (imageStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+func (imageStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
This field is separate from the manifest. We may choose not to allow users
to modify it, but it would be ok if they did.

On Thursday, September 17, 2015, Michal Minar <notifications@github.com>
wrote:

> In pkg/image/registry/image/strategy.go
> <https://github.com/openshift/origin/pull/4666#discussion_r39726448>:
>
> > @@ -44,6 +45,25 @@ func (imageStrategy) AllowCreateOnUpdate() bool {
> >  	return false
> >  }
> >
> > +func (imageStrategy) AllowUnconditionalUpdate() bool {
> > +	return false
> > +}
> > +
> > +// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
> > +func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
>
> I don't think we should allow for updates of DockerImageReference
> containing fully-qualified name, which is part of manifest that can't be
> changed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4666/files#r39726448>.
>